authors/apps/profiles/tests/test_profiles.py

```diff
@@ -0,0 +1,163 @@
+from django.test import TestCase, Client
+from django.urls import reverse
+from rest_framework import status
+
+from ..models import Profile
+
+class TestProfile(TestCase):
+    """
+    Check if profiles functionality is working as expected
+
+    """
+    def setUp(self):
+        client = Client()
+        self.login_url = reverse("auth:login")
+        self.register_url = reverse("auth:register")
+        self.profile_url = reverse("prof:authors_profile")
+    
+
+        self.valid_user_credentials = {
+                "user": {
+                    "username": "Alpha",
+                    "email": "alphaandela@gmail.com",
+                    "password": "@Alpha254"
+                }
+        }
+
+        self.user_credentials = {
+                "user": {
+                    "username": "Alpha",
+                    "email": "alphaandela@gmail.com",
+                    "password": "@Alpha254"
+                }
+        }
+
+        self.update_profile = {
+        'profile':
+        {
+            "bio": "i am a tech enthusiast",
+            "image": "default.jpg",
+            "company": "Andela",
+            "location": "cape Town"
+        }
+        }
+
+    def user_registration(self,data):
+       #register a user
+        response = self.client.post(
+            self.register_url,
+            self.valid_user_credentials,
+            content_type='application/json'
+        )
+        return response
+
+
+    def login(self, data):
+        #login user
+        response = self.client.post(
+            self.login_url,
+            self.valid_user_credentials,
+            content_type='application/json'
+        )
+        return response
+
+
+    def test_model_auto_create_user_profile(self):
+        #test profile gets created when user is registered
+        initial_count = Profile.objects.count()
+        self.user_registration(self.valid_user_credentials) 
+        new_count = Profile.objects.count()
+        self.assertNotEqual(initial_count, new_count)
+
+    def test_get_other_user_profile(self):
+        #test user can see other users profile
+        self.user_registration(self.valid_user_credentials)
+        self.login(self.valid_user_credentials)
+        res = self.login(self.valid_user_credentials)
+        username = res.data['username']
+        token = res.data['token']
+        url = '/api/v1/profiles/' + username + '/'
+        response = self.client.get(
+            url,
+            content_type='application/json',
+            HTTP_AUTHORIZATION='Bearer ' + token
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
Nice work with the test @Ruiru11 however it would be better if you could also assert for response messages and data

Reply:
For all the tests that have a response message, I have tested the response.